ocs_ci/utility/kubevirt_vm.py

```diff
@@ -0,0 +1,248 @@
+# -*- coding: utf8 -*-
+"""
+Module for interactions with IBM Cloud Cluster.
+
+"""
+
+import logging
+import time
+
+from ocs_ci.deployment.helpers.hypershift_base import get_cluster_vm_namespace
+from ocs_ci.framework import config
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.utility.decorators import switch_to_provider_for_function
+from ocs_ci.ocs import constants
+
+
+logger = logging.getLogger(name=__file__)
+
+
+def get_vm_name(vm):
+    """
+    Get the virtual machine name
+
+    Args:
+        vm (dict): The dictionary that represents the virtual machine
+
+    Returns:
+        str: The virtual machine name
+
+    """
+    return vm["metadata"]["name"]
+
+
+class KubevirtVM(object):
+    """
+    Wrapper for VM objects with kubevirt. The class should be used in Provider mode
+    when we have provider and client clusters in the run, and the client is the primary cluster.
+    """
+
+    def __init__(self):
```

Comment:
Shouldn't it be ensured that the constructor is called in provider context? ocp_vm methods like get() work only in provider context, right?

Reply:
Yes, I will add the param cluster_name and ensure it is also called with provider context, as @jilju suggested in his comments.